determined-ai/determined
# fix: charts should show is loading instead of no data [WEB-1367]

## Description

Introduces new `Loadable`s as well as a `hasData`  check in `useTrialMetrics` in order to determine whether data for the charts is available. Now, if the UI is still waiting to load data for any chart in the `ChartGrid` we show a spinner for the individual charts instead of flashing `No data to plot.`  

Current charts without loading state:
https://github.com/determined-ai/determined/assets/103522725/31baad64-7013-4163-93c0-1bde97aea110

New Charts with loading states:
https://github.com/determined-ai/determined/assets/103522725/8ab200c0-0025-4dff-a356-cf6a2813dae1





## Test Plan
1. In the new experiment list view, open the Compare view and the `Metrics` tab.
2. Select an experiment that has metrics, ensure that the charts show a spinner and do not flash `No Data to plot`. Once the charts have loaded they should show the correct data.
3. Select more experiments, ensure that all charts show a spinner while data is being loaded and that `No Data to plot` is not flashed on the screen for any chart that does contain data.
4. De-select experiments and sure that the charts show a spinner while the charts are updating, ensure that `No Data to plot` is not flashed on the screen for any chart that does contain data.





## Commentary (optional)





## Checklist

  See [Release Note](https://github.com/determined-ai/determined/blob/master/docs/release-notes/README.md) for details.

## Ticket
WEB-1367
